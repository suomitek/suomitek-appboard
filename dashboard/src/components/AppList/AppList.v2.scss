.header-button {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  height: 100%;
}

.applist-empty {
  text-align: center;
}
