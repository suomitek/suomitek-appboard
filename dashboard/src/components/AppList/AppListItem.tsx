import * as React from "react";
import helmIcon from "../../icons/helm.svg";
import placeholder from "../../placeholder.png";
import { IAppOverview } from "../../shared/types";
import * as url from "../../shared/url";
import InfoCard from "../InfoCard";
import "./AppListItem.css";

export interface IAppListItemProps {
  app: IAppOverview;
  cluster: string;
}

class AppListItem extends React.Component<IAppListItemProps> {
  public render() {
    const { app, cluster } = this.props;
    const icon = app.icon ? app.icon : placeholder;
    const banner =
      app.updateInfo && !app.updateInfo.error && !app.updateInfo.upToDate
        ? "Update available"
        : undefined;
    return (
      <InfoCard
        key={app.releaseName}
        link={url.app.apps.get(cluster, app.namespace, app.releaseName)}
        title={app.releaseName}
        icon={icon}
        info={`${app.chart} v${app.version || "-"}`}
        banner={banner}
        tag1Content={app.namespace}
        tag2Content={app.status.toLocaleLowerCase()}
        tag2Class={app.status.toLocaleLowerCase()}
        subIcon={helmIcon}
      />
    );
  }
}

export default AppListItem;
