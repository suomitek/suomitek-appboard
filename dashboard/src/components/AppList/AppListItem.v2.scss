.info-icon {
  display: grid;
  align-items: center;
  height: 100%;
  padding-left: 0.4rem;
  justify-content: center;
}

/* stylelint-disable-next-line selector-type-no-unknown */
.is-success cds-icon {
  --color: var(--clr-color-success-700);
}
