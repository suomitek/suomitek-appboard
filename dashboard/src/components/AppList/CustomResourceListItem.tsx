import * as React from "react";

import * as url from "shared/url";
import operatorIcon from "../../icons/operator-framework.svg";
import placeholder from "../../placeholder.png";
import { IClusterServiceVersion, IResource } from "../../shared/types";
import UnexpectedErrorPage from "../ErrorAlert/UnexpectedErrorAlert";
import InfoCard from "../InfoCard";

interface ICustomResourceListItemProps {
  cluster: string;
  resource: IResource;
  csv: IClusterServiceVersion;
}

class CustomResourceListItem extends React.Component<ICustomResourceListItemProps> {
  public render() {
    const { cluster, resource, csv } = this.props;
    const icon = csv.spec.icon
      ? `data:${csv.spec.icon[0].mediatype};base64,${csv.spec.icon[0].base64data}`
      : placeholder;
    const crd = csv.spec.customresourcedefinitions.owned.find(c => c.kind === resource.kind);
    if (!crd) {
      // Unexpected error, CRD should be present if resource is defined
      return (
        <UnexpectedErrorPage
          text={`Unable to retrieve the CustomResourceDefinition for ${resource.kind} in ${csv.metadata.name}`}
        />
      );
    }
    return (
      <InfoCard
        key={resource.metadata.name}
        link={url.app.operatorInstances.view(
          cluster,
          resource.metadata.namespace,
          csv.metadata.name,
          crd.name,
          resource.metadata.name,
        )}
        title={resource.metadata.name}
        icon={icon}
        info={`${resource.kind} v${csv.spec.version || "-"}`}
        tag1Content={resource.metadata.namespace}
        tag2Content={csv.metadata.name.split(".")[0]}
        subIcon={operatorIcon}
      />
    );
  }
}

export default CustomResourceListItem;
