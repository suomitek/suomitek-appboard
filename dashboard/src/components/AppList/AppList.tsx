import * as React from "react";
import { Link } from "react-router-dom";

import { IFeatureFlags } from "shared/Config";
import { IAppOverview, IAppState, IClusterServiceVersion, IResource } from "../../shared/types";
import * as url from "../../shared/url";
import { escapeRegExp } from "../../shared/utils";
import { CardGrid } from "../Card";
import { ErrorSelector, MessageAlert } from "../ErrorAlert";
import LoadingWrapper from "../LoadingWrapper";
import PageHeader from "../PageHeader";
import SearchFilter from "../SearchFilter";
import AppListItem from "./AppListItem";
import CustomResourceListItem from "./CustomResourceListItem";

export interface IAppListProps {
  apps: IAppState;
  fetchAppsWithUpdateInfo: (cluster: string, ns: string, all: boolean) => void;
  cluster: string;
  namespace: string;
  pushSearchFilter: (filter: string) => any;
  filter: string;
  getCustomResources: (ns: string) => void;
  customResources: IResource[];
  isFetchingResources: boolean;
  csvs: IClusterServiceVersion[];
  featureFlags: IFeatureFlags;
}

interface IAppListState {
  filter: string;
}

class AppList extends React.Component<IAppListProps, IAppListState> {
  public state: IAppListState = { filter: "" };
  public componentDidMount() {
    const {
      fetchAppsWithUpdateInfo,
      filter,
      cluster,
      namespace,
      apps,
      getCustomResources,
    } = this.props;
    fetchAppsWithUpdateInfo(cluster, namespace, apps.listingAll);
    if (this.props.featureFlags.operators) {
      getCustomResources(namespace);
    }
    this.setState({ filter });
  }

  public componentDidUpdate(prevProps: IAppListProps) {
    const {
      apps: { error, listingAll },
      fetchAppsWithUpdateInfo,
      getCustomResources,
      filter,
      cluster,
      namespace,
    } = this.props;
    // refetch if new namespace or error removed due to location change
    if (
      prevProps.namespace !== namespace ||
      prevProps.cluster !== cluster ||
      (!error && prevProps.apps.error)
    ) {
      fetchAppsWithUpdateInfo(cluster, namespace, listingAll);
      if (this.props.featureFlags.operators) {
        getCustomResources(namespace);
      }
    }
    if (prevProps.filter !== filter) {
      this.setState({ filter });
    }
  }

  public render() {
    const {
      apps: { error, isFetching },
      cluster,
      isFetchingResources,
      namespace,
    } = this.props;
    return (
      <section className="AppList">
        <PageHeader>
          <div className="col-9">
            <div className="row">
              <h1>Applications</h1>
              {!error && this.appListControls()}
            </div>
          </div>
          <div className="col-3 text-r align-center">
            {!error && (
              <Link to={url.app.catalog(cluster, namespace)}>
                <button className="deploy-button button button-accent">Deploy App</button>
              </Link>
            )}
          </div>
        </PageHeader>
        <main>
          <LoadingWrapper loaded={!isFetching && !isFetchingResources}>
            {error ? (
              <ErrorSelector
                error={error}
                action="list"
                resource="Applications"
                namespace={this.props.namespace}
              />
            ) : (
              this.appListItems()
            )}
          </LoadingWrapper>
        </main>
      </section>
    );
  }

  public appListControls() {
    const {
      pushSearchFilter,
      apps: { listingAll },
    } = this.props;
    return (
      <React.Fragment>
        <SearchFilter
          key="searchFilter"
          className="margin-l-big"
          placeholder="search apps..."
          onChange={this.handleFilterQueryChange}
          value={this.state.filter}
          onSubmit={pushSearchFilter}
        />
        <label className="checkbox margin-r-big margin-l-big margin-t-big" key="listall">
          <input type="checkbox" checked={listingAll} onChange={this.toggleListAll} />
          <span>Show deleted apps</span>
        </label>
      </React.Fragment>
    );
  }

  public appListItems() {
    const {
      apps: { listOverview },
      cluster,
      customResources,
    } = this.props;
    const filteredReleases = this.filteredReleases(listOverview || [], this.state.filter);
    const filteredCRs = this.filteredCRs(customResources, this.state.filter);
    if (filteredReleases.length === 0 && filteredCRs.length === 0) {
      return (
        <MessageAlert header="Supercharge your Kubernetes cluster">
          <div>
            <p className="margin-v-normal">
              Deploy applications on your Kubernetes cluster with a single click.
            </p>
          </div>
        </MessageAlert>
      );
    }
    return (
      <div>
        <CardGrid>
          {filteredReleases.map(r => {
            return <AppListItem key={r.releaseName} app={r} cluster={cluster} />;
          })}
          {filteredCRs.map(r => {
            const csv = this.props.csvs.find(c =>
              c.spec.customresourcedefinitions.owned.some(crd => crd.kind === r.kind),
            );
            return (
              <CustomResourceListItem
                cluster={cluster}
                key={r.metadata.name}
                resource={r}
                csv={csv!}
              />
            );
          })}
        </CardGrid>
      </div>
    );
  }

  private toggleListAll = () => {
    this.props.fetchAppsWithUpdateInfo(
      this.props.cluster,
      this.props.namespace,
      !this.props.apps.listingAll,
    );
  };

  private filteredReleases(apps: IAppOverview[], filter: string) {
    return apps.filter(a => new RegExp(escapeRegExp(filter), "i").test(a.releaseName));
  }

  private filteredCRs(crs: IResource[], filter: string) {
    return crs.filter(cr => new RegExp(escapeRegExp(filter), "i").test(cr.metadata.name));
  }

  private handleFilterQueryChange = (filter: string) => {
    this.setState({
      filter,
    });
  };
}

export default AppList;
