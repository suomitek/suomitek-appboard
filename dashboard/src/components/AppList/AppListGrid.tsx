import * as React from "react";
import { Link } from "react-router-dom";

import { IAppOverview, IClusterServiceVersion, IResource } from "../../shared/types";
import * as url from "../../shared/url";
import { escapeRegExp } from "../../shared/utils";
import CardGrid from "../Card/CardGrid.v2";
import AppListItem from "./AppListItem.v2";

import Alert from "../js/Alert";
import "./AppList.v2.css";
import CustomResourceListItem from "./CustomResourceListItem.v2";

export interface IAppListProps {
  appList: IAppOverview[] | undefined;
  cluster: string;
  namespace: string;
  filter: string;
  customResources: IResource[];
  csvs: IClusterServiceVersion[];
  appVersion: string;
}

function AppListGrid(props: IAppListProps) {
  const { appList, customResources, cluster, namespace, appVersion, filter } = props;
  const filteredReleases = (appList || []).filter(a =>
    new RegExp(escapeRegExp(filter), "i").test(a.releaseName),
  );
  const filteredCRs = customResources.filter(cr =>
    new RegExp(escapeRegExp(filter), "i").test(cr.metadata.name),
  );

  if (filteredReleases.length === 0 && filteredCRs.length === 0) {
    return (
      <div className="applist-empty">
        <Alert>Deploy applications on your Kubernetes cluster with a single click.</Alert>
        <h2>Welcome To Kubeapps</h2>
        <p>
          Start browsing your <Link to={url.app.catalog(cluster, namespace)}>favourite apps</Link>{" "}
          or check the{" "}
          <a
            href={`https://github.com/kubeapps/kubeapps/blob/${appVersion}/docs/user/getting-started.md`}
            target="_blank"
            rel="noopener noreferrer"
          >
            documentation
          </a>
          .
        </p>
      </div>
    );
  }
  return (
    <CardGrid>
      <>
        {filteredReleases.map(r => {
          return <AppListItem key={r.releaseName} app={r} cluster={cluster} />;
        })}
        {filteredCRs.map(r => {
          const csv = props.csvs.find(c =>
            c.spec.customresourcedefinitions.owned.some(crd => crd.kind === r.kind),
          );
          return (
            <CustomResourceListItem
              cluster={cluster}
              key={r.metadata.name}
              resource={r}
              csv={csv!}
            />
          );
        })}
      </>
    </CardGrid>
  );
}

export default AppListGrid;
