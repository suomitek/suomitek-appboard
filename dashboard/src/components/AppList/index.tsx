import * as React from "react";

import { IAppListProps } from "./AppList.v2";

const AppList = React.lazy(() => import("./AppList"));
const AppListV2 = React.lazy(() => import("./AppList.v2"));

interface IAppListSelectorProps extends IAppListProps {
  UI: string;
}

const AppListSelector: React.FC<IAppListSelectorProps> = props => (
  <React.Suspense fallback={null}>
    {props.UI === "clarity" ? <AppListV2 {...props} /> : <AppList {...props} />}
  </React.Suspense>
);

export default AppListSelector;
