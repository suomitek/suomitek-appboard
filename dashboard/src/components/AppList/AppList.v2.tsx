import * as React from "react";
import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { CdsButton, CdsIcon } from "../Clarity/clarity";

import { IFeatureFlags } from "../../shared/Config";
import { IAppState, IClusterServiceVersion, IResource } from "../../shared/types";
import * as url from "../../shared/url";
import Column from "../js/Column";
import PageHeader from "../PageHeader/PageHeader.v2";
import SearchFilter from "../SearchFilter/SearchFilter.v2";

import Alert from "components/js/Alert";
import Row from "components/js/Row";
import LoadingWrapper from "components/LoadingWrapper/LoadingWrapper.v2";
import "./AppList.v2.css";
import AppListGrid from "./AppListGrid";

export interface IAppListProps {
  apps: IAppState;
  fetchAppsWithUpdateInfo: (cluster: string, ns: string, all: boolean) => void;
  cluster: string;
  namespace: string;
  pushSearchFilter: (filter: string) => any;
  filter: string;
  getCustomResources: (ns: string) => void;
  customResources: IResource[];
  isFetchingResources: boolean;
  csvs: IClusterServiceVersion[];
  featureFlags: IFeatureFlags;
  appVersion: string;
}

function AppList(props: IAppListProps) {
  const [filter, setFilter] = useState("");
  const {
    fetchAppsWithUpdateInfo,
    filter: filterProps,
    namespace,
    getCustomResources,
    apps: { error, isFetching, listOverview },
    cluster,
    isFetchingResources,
    pushSearchFilter,
    customResources,
    appVersion,
    csvs,
  } = props;

  useEffect(() => {
    fetchAppsWithUpdateInfo(cluster, namespace, true);
    getCustomResources(namespace);
  }, [cluster, namespace, fetchAppsWithUpdateInfo, getCustomResources]);

  useEffect(() => {
    setFilter(filterProps);
  }, [filterProps]);

  return (
    <section>
      <PageHeader>
        <Column span={10}>
          <Row>
            <h1>Applications</h1>
            <SearchFilter
              key="searchFilter"
              placeholder="search apps..."
              onChange={setFilter}
              value={filter}
              onSubmit={pushSearchFilter}
            />
          </Row>
        </Column>
        <Column span={2}>
          <div className="header-button">
            <Link to={url.app.catalog(cluster, namespace)}>
              <CdsButton status="primary">
                <CdsIcon shape="deploy" inverse={true} /> Deploy
              </CdsButton>
            </Link>
          </div>
        </Column>
      </PageHeader>
      <LoadingWrapper loaded={!isFetching && !isFetchingResources}>
        {error ? (
          <Alert theme="danger">Unable to list apps: {error.message}</Alert>
        ) : (
          <AppListGrid
            appList={listOverview}
            customResources={customResources}
            cluster={cluster}
            namespace={namespace}
            appVersion={appVersion}
            filter={filter}
            csvs={csvs}
          />
        )}
      </LoadingWrapper>
    </section>
  );
}

export default AppList;
