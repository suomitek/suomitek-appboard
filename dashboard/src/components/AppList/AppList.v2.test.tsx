import { mount, shallow } from "enzyme";
import context from "jest-plugin-context";
import * as React from "react";

import SearchFilter from "components/SearchFilter/SearchFilter.v2";
import { BrowserRouter as Router } from "react-router-dom";
import itBehavesLike from "../../shared/specs";
import { IAppOverview, IAppState } from "../../shared/types";
import Alert from "../js/Alert";
import AppList, { IAppListProps } from "./AppList.v2";
import AppListItem from "./AppListItem.v2";
import CustomResourceListItem from "./CustomResourceListItem.v2";

let props = {} as any;

const defaultProps: IAppListProps = {
  apps: {} as IAppState,
  filter: "",
  namespace: "default",
  cluster: "defaultc",
  pushSearchFilter: jest.fn(),
  fetchAppsWithUpdateInfo: jest.fn(),
  getCustomResources: jest.fn(),
  isFetchingResources: false,
  customResources: [],
  csvs: [],
  featureFlags: { operators: true, ui: "hex" },
  appVersion: "v1.0.0",
};

context("when changing props", () => {
  it("should fetch apps in the new namespace", async () => {
    const fetchAppsWithUpdateInfo = jest.fn();
    const getCustomResources = jest.fn();
    mount(
      <Router>
        <AppList
          {...defaultProps}
          fetchAppsWithUpdateInfo={fetchAppsWithUpdateInfo}
          getCustomResources={getCustomResources}
          namespace="foo"
        />
      </Router>,
    );
    expect(fetchAppsWithUpdateInfo).toHaveBeenCalledWith("defaultc", "foo", true);
    expect(getCustomResources).toHaveBeenCalledWith("foo");
  });

  it("should update the filter", () => {
    const wrapper = mount(
      <Router>
        <AppList {...defaultProps} filter="foo" />
      </Router>,
    );
    expect(wrapper.find(SearchFilter).prop("value")).toEqual("foo");
  });
});

context("while fetching apps", () => {
  props = { ...defaultProps, apps: { isFetching: true } };

  itBehavesLike("aLoadingComponent", { component: AppList, props });
  itBehavesLike("aLoadingComponent", {
    component: AppList,
    props: { ...defaultProps, isFetchingResources: true },
  });

  it("renders a Application header", () => {
    const wrapper = shallow(<AppList {...props} />);
    expect(wrapper.find("h1").text()).toContain("Applications");
  });

  it("shows the search filter and deploy button", () => {
    const wrapper = shallow(<AppList {...props} />);
    expect(wrapper.find("SearchFilter")).toExist();
    expect(wrapper.find("Link").findWhere(l => l.text().includes("Deploy"))).toExist();
  });
});

context("when fetched but not apps available", () => {
  beforeEach(() => {
    props = { ...defaultProps, apps: { listOverview: [] } };
  });

  it("matches the snapshot", () => {
    const wrapper = shallow(<AppList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a welcome message", () => {
    const wrapper = mount(
      <Router>
        <AppList {...props} />
      </Router>,
    );
    expect(wrapper.find(".applist-empty").text()).toContain("Welcome To Suomitek-appboard");
  });

  it("shows the search filter and deploy button", () => {
    const wrapper = shallow(<AppList {...props} />);
    expect(wrapper.find("SearchFilter")).toExist();
    expect(wrapper.find("Link").findWhere(l => l.text().includes("Deploy"))).toExist();
  });
});

context("when an error is present", () => {
  beforeEach(() => {
    props = { ...defaultProps, apps: { error: new Error("Boom!") } };
  });

  it("matches the snapshot", () => {
    const wrapper = shallow(<AppList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a generic error message", () => {
    const wrapper = shallow(<AppList {...props} />);
    expect(wrapper.find(Alert)).toExist();
    expect(wrapper.find(Alert).html()).toContain("Boom!");
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a Application header", () => {
    const wrapper = shallow(<AppList {...props} />);
    expect(wrapper.find("h1").text()).toContain("Applications");
  });
});

context("when apps available", () => {
  beforeEach(() => {
    props = {
      ...defaultProps,
      apps: {
        listOverview: [
          {
            releaseName: "foo",
            chartMetadata: {
              name: "bar",
              version: "1.0.0",
              appVersion: "0.1.0",
            },
            status: "deployed",
          } as IAppOverview,
        ],
      },
    };
  });

  it("matches the snapshot", () => {
    const wrapper = shallow(<AppList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a CardGrid with the available Apps", () => {
    const wrapper = mount(
      <Router>
        <AppList {...props} />
      </Router>,
    );
    const itemList = wrapper.find(AppListItem);
    expect(itemList).toExist();
    expect(itemList.key()).toBe("foo");
  });
});

it("filters apps", () => {
  const wrapper = mount(
    <Router>
      <AppList
        {...defaultProps}
        apps={
          {
            isFetching: false,
            items: [],
            listOverview: [
              {
                releaseName: "foo",
                chartMetadata: {
                  name: "foobar",
                  version: "1.0.0",
                  appVersion: "0.1.0",
                },
                status: "deployed",
              } as IAppOverview,
              {
                releaseName: "bar",
                chartMetadata: {
                  name: "foobar",
                  version: "1.0.0",
                  appVersion: "0.1.0",
                },
                status: "deployed",
              } as IAppOverview,
            ],
            listingAll: false,
          } as IAppState
        }
        filter="bar"
      />
      ,
    </Router>,
  );
  expect(wrapper.find(AppListItem).key()).toBe("bar");
});

context("when custom resources available", () => {
  beforeEach(() => {
    const cr = { kind: "KubeappsCluster", metadata: { name: "foo-cluster" } };
    const csv = {
      metadata: {
        name: "foo",
      },
      spec: {
        customresourcedefinitions: {
          owned: [
            {
              kind: "KubeappsCluster",
            },
          ],
        },
      },
    };
    props = {
      ...defaultProps,
      customResources: [cr],
      csvs: [csv],
    };
  });

  it("matches the snapshot", () => {
    const wrapper = shallow(<AppList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a CardGrid with the available resources", () => {
    const wrapper = mount(
      <Router>
        <AppList {...props} />
      </Router>,
    );
    const itemList = wrapper.find(CustomResourceListItem);
    expect(itemList).toExist();
    expect(itemList.key()).toBe("foo-cluster");
  });

  it("filters out items", () => {
    const wrapper = mount(
      <Router>
        <AppList {...props} filter="nop" />
      </Router>,
    );
    const itemList = wrapper.find(CustomResourceListItem);
    expect(itemList).not.toExist();
  });
});
