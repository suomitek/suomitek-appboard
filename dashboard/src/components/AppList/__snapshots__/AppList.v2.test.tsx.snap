// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when an error is present matches the snapshot 1`] = `
<section>
  <PageHeader>
    <Column
      listItem={false}
      span={10}
    >
      <Row
        list={false}
      >
        <h1>
          Applications
        </h1>
        <SearchFilter
          key="searchFilter"
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search apps..."
          value=""
        />
      </Row>
    </Column>
    <Column
      listItem={false}
      span={2}
    >
      <div
        className="header-button"
      >
        <Link
          to="/c/defaultc/ns/default/catalog"
        >
          <ReactWrapperComponent
            status="primary"
          >
            <ReactWrapperComponent
              inverse={true}
              shape="deploy"
            />
             Deploy
          </ReactWrapperComponent>
        </Link>
      </div>
    </Column>
  </PageHeader>
  <LoadingWrapper
    loaded={true}
  >
    <Alert
      app={false}
      compact={false}
      theme="danger"
    >
      Unable to list apps: 
      Boom!
    </Alert>
  </LoadingWrapper>
</section>
`;

exports[`when an error is present renders a generic error message 1`] = `
<section>
  <PageHeader>
    <Column
      listItem={false}
      span={10}
    >
      <Row
        list={false}
      >
        <h1>
          Applications
        </h1>
        <SearchFilter
          key="searchFilter"
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search apps..."
          value=""
        />
      </Row>
    </Column>
    <Column
      listItem={false}
      span={2}
    >
      <div
        className="header-button"
      >
        <Link
          to="/c/defaultc/ns/default/catalog"
        >
          <ReactWrapperComponent
            status="primary"
          >
            <ReactWrapperComponent
              inverse={true}
              shape="deploy"
            />
             Deploy
          </ReactWrapperComponent>
        </Link>
      </div>
    </Column>
  </PageHeader>
  <LoadingWrapper
    loaded={true}
  >
    <Alert
      app={false}
      compact={false}
      theme="danger"
    >
      Unable to list apps: 
      Boom!
    </Alert>
  </LoadingWrapper>
</section>
`;

exports[`when apps available matches the snapshot 1`] = `
<section>
  <PageHeader>
    <Column
      listItem={false}
      span={10}
    >
      <Row
        list={false}
      >
        <h1>
          Applications
        </h1>
        <SearchFilter
          key="searchFilter"
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search apps..."
          value=""
        />
      </Row>
    </Column>
    <Column
      listItem={false}
      span={2}
    >
      <div
        className="header-button"
      >
        <Link
          to="/c/defaultc/ns/default/catalog"
        >
          <ReactWrapperComponent
            status="primary"
          >
            <ReactWrapperComponent
              inverse={true}
              shape="deploy"
            />
             Deploy
          </ReactWrapperComponent>
        </Link>
      </div>
    </Column>
  </PageHeader>
  <LoadingWrapper
    loaded={true}
  >
    <AppListGrid
      appList={
        Array [
          Object {
            "chartMetadata": Object {
              "appVersion": "0.1.0",
              "name": "bar",
              "version": "1.0.0",
            },
            "releaseName": "foo",
            "status": "deployed",
          },
        ]
      }
      appVersion="v1.0.0"
      cluster="defaultc"
      csvs={Array []}
      customResources={Array []}
      filter=""
      namespace="default"
    />
  </LoadingWrapper>
</section>
`;

exports[`when custom resources available matches the snapshot 1`] = `
<section>
  <PageHeader>
    <Column
      listItem={false}
      span={10}
    >
      <Row
        list={false}
      >
        <h1>
          Applications
        </h1>
        <SearchFilter
          key="searchFilter"
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search apps..."
          value=""
        />
      </Row>
    </Column>
    <Column
      listItem={false}
      span={2}
    >
      <div
        className="header-button"
      >
        <Link
          to="/c/defaultc/ns/default/catalog"
        >
          <ReactWrapperComponent
            status="primary"
          >
            <ReactWrapperComponent
              inverse={true}
              shape="deploy"
            />
             Deploy
          </ReactWrapperComponent>
        </Link>
      </div>
    </Column>
  </PageHeader>
  <LoadingWrapper
    loaded={true}
  >
    <AppListGrid
      appVersion="v1.0.0"
      cluster="defaultc"
      csvs={
        Array [
          Object {
            "metadata": Object {
              "name": "foo",
            },
            "spec": Object {
              "customresourcedefinitions": Object {
                "owned": Array [
                  Object {
                    "kind": "KubeappsCluster",
                  },
                ],
              },
            },
          },
        ]
      }
      customResources={
        Array [
          Object {
            "kind": "KubeappsCluster",
            "metadata": Object {
              "name": "foo-cluster",
            },
          },
        ]
      }
      filter=""
      namespace="default"
    />
  </LoadingWrapper>
</section>
`;

exports[`when fetched but not apps available matches the snapshot 1`] = `
<section>
  <PageHeader>
    <Column
      listItem={false}
      span={10}
    >
      <Row
        list={false}
      >
        <h1>
          Applications
        </h1>
        <SearchFilter
          key="searchFilter"
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search apps..."
          value=""
        />
      </Row>
    </Column>
    <Column
      listItem={false}
      span={2}
    >
      <div
        className="header-button"
      >
        <Link
          to="/c/defaultc/ns/default/catalog"
        >
          <ReactWrapperComponent
            status="primary"
          >
            <ReactWrapperComponent
              inverse={true}
              shape="deploy"
            />
             Deploy
          </ReactWrapperComponent>
        </Link>
      </div>
    </Column>
  </PageHeader>
  <LoadingWrapper
    loaded={true}
  >
    <AppListGrid
      appList={Array []}
      appVersion="v1.0.0"
      cluster="defaultc"
      csvs={Array []}
      customResources={Array []}
      filter=""
      namespace="default"
    />
  </LoadingWrapper>
</section>
`;

exports[`while fetching apps loading spinner matches the snapshot 1`] = `
<section>
  <PageHeader>
    <Column
      listItem={false}
      span={10}
    >
      <Row
        list={false}
      >
        <h1>
          Applications
        </h1>
        <SearchFilter
          key="searchFilter"
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search apps..."
          value=""
        />
      </Row>
    </Column>
    <Column
      listItem={false}
      span={2}
    >
      <div
        className="header-button"
      >
        <Link
          to="/c/defaultc/ns/default/catalog"
        >
          <ReactWrapperComponent
            status="primary"
          >
            <ReactWrapperComponent
              inverse={true}
              shape="deploy"
            />
             Deploy
          </ReactWrapperComponent>
        </Link>
      </div>
    </Column>
  </PageHeader>
  <LoadingWrapper
    loaded={false}
  >
    <AppListGrid
      appVersion="v1.0.0"
      cluster="defaultc"
      csvs={Array []}
      customResources={Array []}
      filter=""
      namespace="default"
    />
  </LoadingWrapper>
</section>
`;

exports[`while fetching apps loading spinner matches the snapshot 2`] = `
<section>
  <PageHeader>
    <Column
      listItem={false}
      span={10}
    >
      <Row
        list={false}
      >
        <h1>
          Applications
        </h1>
        <SearchFilter
          key="searchFilter"
          onChange={[Function]}
          onSubmit={[MockFunction]}
          placeholder="search apps..."
          value=""
        />
      </Row>
    </Column>
    <Column
      listItem={false}
      span={2}
    >
      <div
        className="header-button"
      >
        <Link
          to="/c/defaultc/ns/default/catalog"
        >
          <ReactWrapperComponent
            status="primary"
          >
            <ReactWrapperComponent
              inverse={true}
              shape="deploy"
            />
             Deploy
          </ReactWrapperComponent>
        </Link>
      </div>
    </Column>
  </PageHeader>
  <LoadingWrapper
    loaded={false}
  >
    <AppListGrid
      appVersion="v1.0.0"
      cluster="defaultc"
      csvs={Array []}
      customResources={Array []}
      filter=""
      namespace="default"
    />
  </LoadingWrapper>
</section>
`;
